htlcswitch/link.go

```diff
@@ -1457,6 +1500,9 @@ func (l *channelLink) htlcManager() {
 				)
 			}
 
+		case runQuery := <-l.stateQueries:
+			runQuery()
```

Comment:
Really like that we are using go chans to handle queries, and I wanna know the design choice behind this, cause alternatively we could just define query types and add a `handleQueries` to avoid letting the caller construct the queries, as it seems to be the opposite of "APIs should be easy to use and hard to misuse".

Reply:
That was considered. I opted for this because it's a bit more general. It's true that this makes the query creator responsible for the correct construction of the query.

Since go doesn't support sum types, it makes the explicit query cases a bit tougher to define, but if we see substantial value in constraining it this way we can rework it to create explicit query types.

If we go that route we need to answer the question of whether we want a product of channels or a sum of queries. Either we need to pipe all the queries through a single channel which would mean we need to unify the query types under a single sum type (which in go we model as an interface with a private method), OR we can have a separate go channel for every query we run and then select on all of them.

Reply:
I guess it depends on how many different types of queries we wanna build.

Reply:
I think we progressively want to move all of the current locking methods to this model of request sent into the event loop.

Reply:
cool then I'd prefer "the single channel" approach, but let's keep the `runInEventLoop` for this iteration to not grow the scope here.